pootle/apps/pootle_statistics/models.py

```diff
@@ -320,10 +320,18 @@ def get_submission_info(self):
     def save(self, *args, **kwargs):
         super(Submission, self).save(*args, **kwargs)
 
-        if self.needs_scorelog():
-            for score in ScoreLog.get_scorelogs(submission=self):
-                if 'action_code' in score and score['user'] is not None:
-                    ScoreLog.objects.create(**score)
+        if not self.needs_scorelog():
+            return
+
+        scorelogs_created = []
+        for score in ScoreLog.get_scorelogs(submission=self):
+            if 'action_code' in score and score['user'] is not None:
+                if not isinstance(score["user"], models.Model):
+                    score["user_id"] = score["user"]
+                    del score["user"]
```

Comment:
This looks like handling a corrupt data situation. Shouldn't it be done in a data migration?